typescript/sdk/src/metadata/warpRouteConfig.ts

```diff
@@ -0,0 +1,27 @@
+import { z } from 'zod';
+
+import { ProtocolType } from '@hyperlane-xyz/utils';
+
+import { TokenType } from '../token/config';
+import { ChainMap } from '../types';
+
+const TokenConfigSchema = z.object({
```

Comment:
What are the advantages of this new schema over the existing CLI one? 
https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/main/typescript/cli/src/config/warp.ts

If this is preferable, should the CLI use it?

Reply:
I mentioned it in the description:

> The chainNames is preferred to base and synthetic bc of its uniqueness as a key and consistency with the rest of the tooling. (eg. injective and inevm has native types on both side where it's not obvious which should be base or synthetic)